crates/fsrt/src/main.rs

```diff
@@ -99,7 +103,69 @@ impl fmt::Display for Error {
     }
 }
 
-impl std::error::Error for Error {}
+fn check_graphql_and_perms(val: &Value) -> Vec<&str> {
+    let mut operations = vec![];
+    match val {
+        Value::Const(Const::Literal(s)) => operations.extend_from_slice(&parse_graphql(s)),
+        Value::Phi(vals) => vals.iter().for_each(|val| match val {
+            Const::Literal(s) => operations.extend_from_slice(&parse_graphql(s)),
+        }),
+        _ => {}
+    }
+    // TODO : Build out permission resolver here
+
+    let permissions_resolver: HashMap<(&str, &str), &str> =
+        [(("compass", "searchTeams"), "read:component:compass")]
+            .iter()
+            .cloned()
+            .collect();
+
+    operations
+        .iter()
+        .filter(|f| permissions_resolver.contains_key(f))
+        .map(|f| permissions_resolver.get(f).unwrap().to_owned())
+        .collect()
+}
+
+fn parse_graphql(s: &str) -> Vec<(&str, &str)> {
+    let mut operations = vec![];
+    if let std::result::Result::Ok(doc) = parse_query::<&str>(s) {
+        doc.definitions.into_iter().for_each(|operation| {
+            if let Definition::Operation(op) = operation {
+                match op {
+                    OperationDefinition::Mutation(mutation) => {
```

Comment:
You can also have plain selection sets as well that need to be parsed.

Reply:
Does AGG allow this ?

Reply:
Pretty sure it did last time I checked, there was just a warning in the UI.

Reply:
Hmmmm... I got an error when I tried, but I added support anyway.... doesn't hurt to.